engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -36,17 +36,20 @@ class EngineJobExecutionActor(replyTo: ActorRef,
 
   override val workflowId = executionData.workflowDescriptor.id
 
-  val jobTag = s"${workflowId.shortString}:${jobKey.call.fullyQualifiedName}:${jobKey.index.fromIndex}:${jobKey.attempt}"
+  val jobTag = s"${workflowId.shortString}:${jobDescriptorKey.call.fullyQualifiedName}:${jobDescriptorKey.index.fromIndex}:${jobDescriptorKey.attempt}"
   val tag = s"EJEA_$jobTag"
 
   // The code that deals with the call cache copying actor only runs if call cache reading is turned on, which it won't be
   // if there isn't a call cache copying actor available. i.e. no code should try to read this variable if this variable
   // isn't set, so while this could be wrapped in an `Option` that would only mask a violation of that invariant.
-  private var callCachingActorProps: Props = _
+  private var backendCacheHitCopyingActorProps: Props = _
 
-  // There's no need to check for a cache hit again if we got preempted
+  // There's no need to check for a cache hit again if we got preempted, or if there's no result copying actor defined
   // NB: this can also change (e.g. if we have a HashError we just force this to CallCachingOff)
-  private var effectiveCallCachingMode = if (jobKey.attempt > 1) callCachingMode.withoutRead else callCachingMode
+  private var effectiveCallCachingMode = if (factory.cacheHitCopyingActorProps.isEmpty || jobDescriptorKey.attempt > 1) callCachingMode.withoutRead else callCachingMode
+
+  // For tests:
+  private[execution] def checkEffectiveCallCachingMode = effectiveCallCachingMode
```

Comment:
Wait, how do you mean? If it's a def you can't update the var from outside the actor?